From bc1237d6da125b620d832ab53a097498cd908dd9 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 23 Feb 2024 18:38:47 +0100
Subject: [PATCH 0665/1695] dts: stm32mp15: describe cpu power supply on
 stm32mp15 boards

Describe cpu power supply on stm32mp15 boards that
is used for CPU OPP.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6200036fed2d90167e9b4a9d99d5dd3934fb369e dts: stm32mp15: describe cpu power supply on stm32mp15 boards]
Change-Id: I6e91746a0880f20d82e02563585faacd8ecf72b4
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362691
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 8 ++++++++
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 8 ++++++++
 2 files changed, 16 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 5f6bf0005..b8a54c748 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -140,6 +140,14 @@
 	st,digbypass;
 };
 
+&cpu0{
+	cpu-supply = <&vddcore>;
+};
+
+&cpu1{
+	cpu-supply = <&vddcore>;
+};
+
 &crc1 {
 	status = "disabled";
 };
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index 7e3113787..7ac634607 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -261,6 +261,14 @@
 	st,digbypass;
 };
 
+&cpu0{
+	cpu-supply = <&vddcore>;
+};
+
+&cpu1{
+	cpu-supply = <&vddcore>;
+};
+
 &i2c4 {
 	compatible = "st,stm32mp15-i2c-non-secure";
 	pinctrl-names = "default", "sleep";
-- 
2.47.2

